internal/cli/atlas/deployments/search/indexes/create.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package indexes
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/atlas/deployments/options"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/atlas/search"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/log"
+	"github.com/mongodb/mongodb-atlas-cli/internal/mongodbclient"
+	"github.com/mongodb/mongodb-atlas-cli/internal/podman"
+	"github.com/mongodb/mongodb-atlas-cli/internal/telemetry"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+const (
+	namePattern        = "^[a-zA-Z0-9][a-zA-Z0-9-]*$"
```

Comment:
where did we get this regex?

Reply:
had it for index name originally (but I probably misread the docs, as this was for the clusterName).
1. searchIndexName - no restrictions are documented and also I tried various chars without getting any error from mongot
2. db & collection names: these have some restrictions, but are not different depending on the environment: [naming-restrictions](https://www.mongodb.com/docs/manual/reference/limits/?_ga=2.64724971.575189893.1694435932-1457532170.1688466502#naming-restrictions)

As these validations are handled at the (local) db level, maybe it's better to just check if these values are not empty in the CLI and leave the validation happen in mongod/mongot, wdyt @gssbzn?

Reply:
I would suggest relying on the local validation on the db so that we don't need to update this regex if something changes on thedb side.

Reply:
I removed hose validations